gym/envs/jax_toy_text/jax_blackjack.py

```diff
@@ -0,0 +1,463 @@
+
+        self.render_mode = render_mode
+        self.renderer = Renderer(self.render_mode, self._render)
+
+    @partial(jit, static_argnums=(0,))
+    def jit_step(self, env_state, action):
+
+        env_state = jax.lax.cond(action, take, notake, env_state)
+
+        state, key = env_state
+        dealer_hand = state[0]
+        player_hand = state[1]
+        dealer_cards = state[2]
+        player_cards = state[3]
+
+        # -1 reward if the player busts, otherwise +1 if better than dealer, 0 if tie, -1 if loss.
+        reward = (is_bust(player_hand) * -1 * action) + (
+            (jnp.logical_not(action)) * cmp(score(player_hand), score(dealer_hand))
+        )
+
+        # in the natural setting, if the player wins with a natural blackjack, then reward is 1.5
+        if self.natural and not self.sutton_and_barto:
+            condition = jnp.logical_and(is_natural(player_hand), (reward == 1))
+            reward = reward * jnp.logical_not(condition) + 1.5 * condition
+
+        # in the sutton and barto setting, if the player gets a natural blackjack and the dealer gets
+        # a non-natural blackjack, the player wins. A dealer natural blackjack and a player
+        # non-natural blackjack should result in a tie.
+        if self.sutton_and_barto:
+            condition = jnp.logical_and(
+                is_natural(player_hand), jnp.logical_not(is_natural(dealer_hand))
+            )
+            reward = reward * jnp.logical_not(condition) + 1 * condition
+
+        # note that only a bust or player action ends the round, the player
+        # can still request another card with 21 cards
+        done = (is_bust(player_hand) * action) + ((jnp.logical_not(action)) * 1)
+
+        new_state = (dealer_hand, player_hand, dealer_cards, player_cards), key
+
+        return new_state, _get_obsv(new_state), reward, done, {}
+
+    @partial(jit, static_argnums=(0,))
+    def jit_reset(self, key):
+        # env_state = self._reset(key)
+
+        player_hand = jnp.zeros(21)
+        dealer_hand = jnp.zeros(21)
+        player_hand, key = draw_hand(key, player_hand)
+        dealer_hand, key = draw_hand(key, dealer_hand)
+        dealer_cards = 2
+        player_cards = 2
+
+        state = (dealer_hand, player_hand, dealer_cards, player_cards)
+
+        key = random.split(key)[0]
+
+        env_state = (state, key)
+
+        return env_state, _get_obsv(env_state)
+
+    def step(self, action):
+        assert self.action_space.contains(action)
+        new_state, observation, reward, done, info = self.jit_step(self.state, action)
+        self.state = new_state
+        self.renderer.render_step()
+        return obs_from_device(observation), float(reward), bool(done), info
+
+    def reset(
+        self,
+        seed: Optional[int] = 0,
+        return_info: bool = False,
+        options: Optional[dict] = None,
+    ):
+        super().reset(seed=seed)
+
+        key = random.PRNGKey(seed)
```

Comment:
Won't this make it so that `env.reset()` always resets to the same initial state unless you explicitly pass a new random seed every time?

Reply:
This should behave the same way as Blackjack-v1 now with respect to seed behavior, barring any mistakes.